### PR TITLE
Upgrade JS deps (Babel, React, Redux, Redux-Form)

### LINKED ISSUE #22: Upgrade to React 0.14 + Toolchain
React has a new release. So does Babeljs. We should upgrade to them. Afaik we shouldn't have too much trouble as we didn't have any warnings prior – but it still might be a task with a unforseeable longtail.

### PR DESCRIPTION
**Pull request by ligthyear**
This PR updates the Javascript Frontend Dependencies and fixes
the errors that followed. Namely it upgrades:

 - babel to v6
 - react to 0.14
 - redux to v3 (and react-redux to v4)
 - redux-form v3.1
 - prosemirror to latest

Fortunately not too many changes were needed in our code base.

This PR fixes #22 .

### CODE REVIEW

**Comment by ligthyear** [comment]
@ezmiller ready for review and merging!

File: beavy/jsbeavy/main.jsx [hunk 1]
```diff
@@ -38,5 +39,5 @@ if (window.PRELOAD.CURRENT_USER) initData.CURRENT_USER = window.PRELOAD.CURRENT_
 const store  = configureStore(initData);
 
 
-React.render(<Root application={Application}
+ReactDom.render(<Root application={Application}
```
Comment (ezmiller) [inline beavy/jsbeavy/main.jsx]: This is interesting. What's 'ReactDom'?

**Comment by ezmiller** [comment]
@homu +r

File: beavy/jsbeavy/main.jsx (hunk 1 above)
Comment (ligthyear) [inline beavy/jsbeavy/main.jsx]: React with its release 0.14 has officially been split into react + different renderers (for the DOM, for native, the canvas and so on). See the release notes to learn more: https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html

**Comment by ezmiller** [comment]
@ligthyear, looked this over and it all looks good to me. nothing in particular jumped out. should i merge? what's the homu command again?

**Comment by ligthyear** [comment]
@ezmiller almost, just a typo: @`homu r+` => runs the tests, merges if all goes fine.

**Comment by ezmiller** [comment]
okay here goes: @homu r+

**Comment by ligthyear** [comment]
@homu retry

**Comment by ezmiller** [comment]
Why did if fail the first time ?

**Comment by ligthyear** [comment]
@ezmiller 

> this was just a travis caching issue.

**Comment by ezmiller** [comment]
Hey @ligthyear after rebasing my branch on upstream/master and running `vagrant up --provision`, I'm getting an error:

hbvlo

I thought it might be related here, but I didn't see any specific changes related to chromium-browser... Ideas?

**Comment by ligthyear** [comment]
The "improve Dev environment" fixes that. Looks like Debian renamed the
package. You can:

Either open your vagrant file, and change "chromium-browser" to "chromium"
manually (please don't commit!) and run vagrant or review and merge the PR
and rebase on top of that.

**Pull request merged**